src/cascadia/TerminalSettingsEditor/LaunchViewModel.cpp

```diff
@@ -23,6 +27,118 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+
+        winrt::hstring result;
+
+        if (UseDefaultLaunchPosition())
+        {
+            result = fmt::format(L"{}, {}, {}", launchModeString, RS_(L"Globals_LaunchModeDefault/Content"), centerOnLaunchString);
+        }
+        else
+        {
+            const std::wstring xPosString = isnan(InitialPosX()) ? RS_(L"Globals_LaunchModeDefault/Content").c_str() : std::to_wstring(gsl::narrow_cast<int>(InitialPosX()));
+            const std::wstring yPosString = isnan(InitialPosY()) ? RS_(L"Globals_LaunchModeDefault/Content").c_str() : std::to_wstring(gsl::narrow_cast<int>(InitialPosY()));
+            result = fmt::format(L"{}, ({},{}), {}", launchModeString, xPosString, yPosString, centerOnLaunchString);
+        }
+        return result;
+    }
+
+    double LaunchViewModel::InitialPosX()
+    {
+        const auto x = _Settings.GlobalSettings().InitialPosition().X;
+        // If there's no value here, return NAN - XAML will ignore it and
+        // put the placeholder text in the box instead
+        const auto xCoord = x.try_as<int64_t>();
+        return xCoord.has_value() ? gsl::narrow_cast<double>(xCoord.value()) : NAN;
+    }
+
+    double LaunchViewModel::InitialPosY()
+    {
+        const auto y = _Settings.GlobalSettings().InitialPosition().Y;
+        // If there's no value here, return NAN - XAML will ignore it and
+        // put the placeholder text in the box instead
+        const auto yCoord = y.try_as<int64_t>();
+        return yCoord.has_value() ? gsl::narrow_cast<double>(yCoord.value()) : NAN;
+    }
+
+    void LaunchViewModel::InitialPosX(double xCoord)
+    {
+        winrt::Windows::Foundation::IReference<int64_t> xCoordRef;
+        // If the value was cleared, xCoord will be NAN, so check for that
+        if (!isnan(xCoord))
+        {
+            xCoordRef = gsl::narrow_cast<int64_t>(xCoord);
+        }
+        const LaunchPosition newPos{ xCoordRef, _Settings.GlobalSettings().InitialPosition().Y };
+        _Settings.GlobalSettings().InitialPosition(newPos);
+        _NotifyChanges(L"LaunchParametersCurrentValue");
+    }
+
+    void LaunchViewModel::InitialPosY(double yCoord)
+    {
+        winrt::Windows::Foundation::IReference<int64_t> yCoordRef;
+        // If the value was cleared, yCoord will be NAN, so check for that
+        if (!isnan(yCoord))
+        {
+            yCoordRef = gsl::narrow_cast<int64_t>(yCoord);
+        }
+        const LaunchPosition newPos{ _Settings.GlobalSettings().InitialPosition().X, yCoordRef };
+        _Settings.GlobalSettings().InitialPosition(newPos);
+        _NotifyChanges(L"LaunchParametersCurrentValue");
+    }
+
+    void LaunchViewModel::UseDefaultLaunchPosition(bool useDefaultPosition)
+    {
+        _useDefaultLaunchPosition = useDefaultPosition;
+        if (useDefaultPosition)
+        {
+            InitialPosX(NAN);
+            InitialPosY(NAN);
+            _NotifyChanges(L"InitialPosX", L"InitialPosY");

```

Comment:
This will generate up to _three_ notifications for `LaunchParametersCurrentValue`, right? Because it will cascade through the `InitialPos#` setters as well.